### PR TITLE
feat: use Halo2 compatible dependencies

### PR DESCRIPTION
**Pull request by DrPeterVanNostrand**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,12 +10,12 @@ license = "MIT/Apache-2.0"
 repository = "https://github.com/filecoin-project/ff-cl-gen"
 
 [dependencies]
-ff = { version = "0.3.0", package = "fff" }
+ff = "0.10.0"
 itertools = { version = "0.8.0" }
 num-bigint = "0.2"
 
 [dev-dependencies]
+blstrs = { git = "https://github.com/filecoin-project/blstrs", branch = "halo" }
```
Comment (DrPeterVanNostrand) [inline Cargo.toml:18]: This should be changed before merging.

**Review by DrPeterVanNostrand — COMMENTED** [review]
Cargo.toml's should be updated before merging.

**Review by dignifiedquire — APPROVED** [review]

File: Cargo.toml [hunk 2]
```diff
@@ -10,12 +10,14 @@ license = "MIT/Apache-2.0"
...
-rand = "0.7"
-lazy_static = "1.2"
+[workspace]
+members = [
+  "gen-test"
+]
+
+[patch."https://github.com/filecoin-project/ff-cl-gen".ff-cl-gen]
+path = "./"
+features = ["gpu"]
```
Comment (vmx) [inline Cargo.toml:23]: Is this in order to enable the `gpu` feature in `gen-test`? I guess enabling the feature there won't work?

Can a crate with `patch` section be published?

File: src/cl/ec.cl [hunk 3]
```diff
@@ -0,0 +1,123 @@
+// Elliptic curve operations (Short Weierstrass Jacobian form)
+
+#define POINT_ZERO ((POINT_projective){FIELD_ZERO, FIELD_ONE, FIELD_ZERO})
+
+typedef struct {
+  FIELD x;
+  FIELD y;
+  #if FIELD_LIMB_BITS == 32
+    uint _padding;
+  #endif
```
Comment (vmx) [inline src/cl/ec.cl:10]: This should be removed. That's only needed for `pairing` and not for `blst`. See https://github.com/filecoin-project/bellperson/pull/211 for more information.

**Review by vmx — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 2 above)
Comment (dignifiedquire) [reply Cargo.toml:23]: No, this needs to go away in some form before publishing. This is a hack around the circular dependency issue

File: src/cl/ec.cl (hunk 3 above)
Comment (dignifiedquire) [reply src/cl/ec.cl]: done

**Review by vmx — APPROVED** [review]
Code lgtm.

**Review by vmx — COMMENTED** [review]
I reviewed the new changes. Looks good to me (I assume things have just moved around, without any functional changes, I haven't checked though). The CI is genuinely failing.

**Pull request merged**